swap_meet/vendor.py

```diff
@@ -1,2 +1,69 @@
+
+from swap_meet.item import Item
+
 class Vendor:
-    pass
\ No newline at end of file
+
+    def __init__(self, inventory=None):
+        if inventory is None:
+            inventory = []
+        self.inventory = inventory
+        
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        try:
+            self.inventory.remove(item)
+            return item
+        except Exception:
+            return False
+
+    def get_by_category(self, category):
+        items_by_category = []
+
+        for item in self.inventory:
+            if item.category == category:
+                items_by_category.append(item)
+        
+        return items_by_category
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
```

Comment:
👍 

Great use of a guard clause to make sure that both vendors are in a valid configuration for performing the swap.

We _could_ still try to use error handling here, but with the two separate `remove` calls, if the second one failed, we would need to to additional work to restore the condition of the inventories, or take a different approach.